backend/indexer/indexer.py

```diff
@@ -364,8 +372,8 @@ async def ingest_data(request: IngestDataToCollectionDto):
                         raise_error_on_failure=created_data_ingestion_run.raise_error_on_failure,
                         batch_size=request.batch_size,
                     )
-                )
-                created_data_ingestion_run.status = DataIngestionRunStatus.COMPLETED
+                Process(target=sync_data_source_to_collection, args=(ingestion_config, )).start()
```

Comment:
While this can work, there is no manager for this process and can run amok.
It would be better to rather use starlette's run_in_threadpool to start with

Reply:
or current loop's run_in_executor with a ProcessPoolExecutor

Reply:
However both of these will require to await and block the request from finishing.
So, it might be better to start with 
https://fastapi.tiangolo.com/tutorial/background-tasks/
https://www.starlette.io/background/

Long term we want to move to proper background workers Celery/Dramatiq/Temporal/TrueFoundry Asycn Service

Reply:
@chiragjn looking at the source for background tasks, it will just await the task on the same event loop, even if we pass a synchronous function, it runs it in a thread pool and "awaits" completion

Reply:
@chiragjn other people have had similar challenge with BackgroundTasks. https://stackoverflow.com/questions/76328904/is-there-a-difference-between-starlette-fastapi-background-tasks-and-simply-usin

Reply:
Then simplest way is to add a global ProcessPoolExecutor with fixed set of n workers - create it in the lifespan context manager of fastapi and terminate/shutdown it in the same lifespan context manager

Submit to the pool and return immediately with `201` no need to wait for indexing to finish, the status will be updated anyway via db